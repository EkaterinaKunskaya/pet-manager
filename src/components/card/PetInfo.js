import React from 'react';
import { Avatar } from 'antd';

import pawImg from '../../assets/image/paw.png';
import { showAge } from '../helpers/ShowAge';

import '../../styles/PetInfo.css';


const PetInfo = ({ card }) => {
  if (card) {
    return (
      <div className='PetInfo'>
        <div className="info-block">
          <div className="img-name">
            <div className="pet-name">{card.petName}</div>
            {(card.img) && <Avatar size={200} src={card.img} />}
            {(!card.img) && <Avatar size={200} src={pawImg} />}

          </div>
          <div className="info-text">
            <p><b>Вид: </b>{card.view}</p>
            <p><b>Порода: </b>{card.breed}</p>
            <p><b>Дата рождения: </b>{card.birthday}</p>
            <p><b>Количество лет: </b>{showAge(card.birthday)}</p>
            <p><b>Пол: </b>{card.sex}</p>
            <p><b>Сведения о репродукции: </b>{card.reproduction}</p>
            <p><b>Номер микрочипа: </b>{card.microchipNumber}</p>
            <p><b>Номер клейма: </b>{card.stampNumber}</p>
            {card.notes && <p><b>Заметки: </b>{card.notes}</p>}
          </div>
        </div>
      </div>
    );
  }
};
export default React.memo(PetInfo);