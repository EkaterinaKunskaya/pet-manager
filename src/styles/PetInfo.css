.PetInfo {
    background: white;
    margin-top: -16px;
    border-radius: 0 8px 8px 8px;
    padding: 50px;
}

.info-block {
    width: 90%;
    margin: 0 auto;
}

.img-name {
    float: left;
    padding: 0 15% 8% 0;
}

.pet-name {
    text-align: center;
    font-family: 'Pacifico', sans-serif;
    font-size: 1.3rem;
    margin-bottom: 20px;
}

.info-text {
    width: 100%;
    font-family: sans-serif;
    font-size: .7rem;
    line-height: 50px;
    word-wrap: break-word;
}


@media (max-width: 420px) {
    
    .PetInfo {
        background: white;
        margin-top: -16px;
        padding: 20px;
    }

    .info-block {
        width: 90%;
        margin: 0 auto;
    }

    .img-name {
        padding: 0 15% 8% 0;
    }

    .pet-name {
        font-size: 1.3rem;
        margin-bottom: 20px;
    }

    .info-text {
        width: 100%;
        font-size: 1.2rem;
        line-height: 40px;
    }
}