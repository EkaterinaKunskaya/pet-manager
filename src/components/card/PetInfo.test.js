import React from 'react';
import renderer from 'react-test-renderer';

import { PetInfo } from './PetInfo';

test('работа PetInfo', () => {

  const component = renderer.create(
    <PetInfo card={{
        petName: 'Милка', 
        view: 'Собака', 
        breed: 'Беспородная',
        birthday: '23.11.2011',
        sex: 'Самка',
        reproduction: 'Стерилизована',
        microchipNumber: '-',
        stampNumber: '-',
        notes: 'Хорошая девочка.',
        img: ''
    }} />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  
});




                                
              

