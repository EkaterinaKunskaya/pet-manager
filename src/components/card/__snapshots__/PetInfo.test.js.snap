// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа PetInfo 1`] = `
<div
  className="PetInfo"
>
  <div
    className="info-block"
  >
    <div
      className="img-name"
    >
      <div
        className="pet-name"
      >
        Милка
      </div>
      <span
        className="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-1s3dcof"
        style={
          Object {
            "fontSize": 18,
            "height": 200,
            "lineHeight": "200px",
            "width": 200,
          }
        }
      >
        <img
          onError={[Function]}
          src="paw.png"
        />
      </span>
    </div>
    <div
      className="info-text"
    >
      <p>
        <b>
          Вид: 
        </b>
        Собака
      </p>
      <p>
        <b>
          Порода: 
        </b>
        Беспородная
      </p>
      <p>
        <b>
          Дата рождения: 
        </b>
        23.11.2011
      </p>
      <p>
        <b>
          Количество лет: 
        </b>
        11 лет 
  и 0 месяцев
      </p>
      <p>
        <b>
          Пол: 
        </b>
        Самка
      </p>
      <p>
        <b>
          Сведения о репродукции: 
        </b>
        Стерилизована
      </p>
      <p>
        <b>
          Номер микрочипа: 
        </b>
        -
      </p>
      <p>
        <b>
          Номер клейма: 
        </b>
        -
      </p>
      <p>
        <b>
          Заметки: 
        </b>
        Хорошая девочка.
      </p>
    </div>
  </div>
</div>
`;

exports[`работа PetInfo 2`] = `
<div
  className="PetInfo"
>
  <div
    className="info-block"
  >
    <div
      className="img-name"
    >
      <div
        className="pet-name"
      >
        Милка
      </div>
      <span
        className="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-1s3dcof"
        style={
          Object {
            "fontSize": 18,
            "height": 200,
            "lineHeight": "200px",
            "width": 200,
          }
        }
      >
        <img
          onError={[Function]}
          src="paw.png"
        />
      </span>
    </div>
    <div
      className="info-text"
    >
      <p>
        <b>
          Вид: 
        </b>
        Собака
      </p>
      <p>
        <b>
          Порода: 
        </b>
        Беспородная
      </p>
      <p>
        <b>
          Дата рождения: 
        </b>
        23.11.2011
      </p>
      <p>
        <b>
          Количество лет: 
        </b>
        11 лет 
  и 0 месяцев
      </p>
      <p>
        <b>
          Пол: 
        </b>
        Самка
      </p>
      <p>
        <b>
          Сведения о репродукции: 
        </b>
        Стерилизована
      </p>
      <p>
        <b>
          Номер микрочипа: 
        </b>
        -
      </p>
      <p>
        <b>
          Номер клейма: 
        </b>
        -
      </p>
      <p>
        <b>
          Заметки: 
        </b>
        Хорошая девочка.
      </p>
    </div>
  </div>
</div>
`;
